doc/scripts/launch_hard_rt_vm.sh

```diff
@@ -0,0 +1,203 @@
+    dev_dev=$(echo $dev_bdf|sed "s/.*://g"|sed "s/\..*//g")
+    dev_fun=$(echo $dev_bdf|sed "s/.*\.//g")
+    dev_drv=$(lspci -k -s $dev_bdf | grep "Kernel driver in use"|sed "s/.*: //g")
+    dev_passthru_opt="-s $guest_pci_nr,passthru,$dev_bus/$dev_dev/$dev_fun"
+
+    echo "passthrough a pci device:"
+    echo dev_info:$dev_info
+    echo dev_path:$dev_path
+    echo dev_vendor_id:$dev_vendor_id
+    echo dev_device_id:$dev_device_id
+    echo dev_bus:$dev_bus
+    echo dev_dev:$dev_dev
+    echo dev_fun:$dev_fun
+    echo dev_drv:$dev_drv
+    echo dev_passthru_opt:$dev_passthru_opt
+
+    # replace its driver with pci-stub
+    echo "$dev_vendor_id $dev_device_id" > /sys/bus/pci/drivers/pci-stub/new_id
+    echo $dev_bdf > $dev_path/driver/unbind
+    echo $dev_bdf > /sys/bus/pci/drivers/pci-stub/bind
+
+    # add to global options
+    dev_opts="$dev_opts $dev_passthru_opt"
+
+    # register for de-passthrough later
+    passthru_devs[$dev_bdf]=$dev_drv
+
+    # increase virtual pci slot number
+    guest_pci_nr=$(expr $guest_pci_nr + 1)
+  done
+}
+
+function de_passthru_pci_devs()
+{
+  for dev_bdf in "${!passthru_devs[@]}"
+  do
+    echo $dev_bdf > /sys/bus/pci/drivers/pci-stub/unbind
+    echo $dev_bdf > /sys/bus/pci/drivers/${passthru_devs[$dev_bdf]}/bind
+  done
+}
+
+# restore pci device driver on exit
+trap de_passthru_pci_devs EXIT
+
+eth=
+disk=
+img=
+passthrus=()
+while getopts hn:b:f: opt
+do
+  case "${opt}" in
+    h)
+        usage;
+        ;;
+    n)
+        eth=${OPTARG}
+        passthrus+=($eth)
+        ;;
+    b)
+        disk="${OPTARG}"
+        passthrus+=($disk)
+        ;;
+    f)
+        img="${OPTARG}"
+        ;;
+    ?)
+        echo "{OPTARG}"
+        ;;
+  esac
+done
+
+if [ ! -z $img ]; then
+  vdisk=/dev/vda3
+  dev_opts="$dev_opts -s $guest_pci_nr,virtio-blk,$img"
+  guest_pci_nr=$(expr $guest_pci_nr + 1)
+elif [ ! -z $disk ];then
+  # detect partition information
+  disk_part=$(basename $disk)
+  vdisk=/dev/$disk_part
+fi
+
+# create a unique tap device for each VM
+if [ -z $eth ];then
+  tap=tap_rtvm
+  tap_exist=$(ip a | grep "$tap" | awk '{print $1}')
+  if [ "$tap_exist"x != "x" ]; then
+    echo "tap device existed, reuse $tap"
+  else
+    ip tuntap add dev $tap mode tap
+    ip link set "$tap" master acrn-br0
+    ip link set dev "$tap" down
+    ip link set dev "$tap" up
+  fi
+
+  dev_opts="$dev_opts -s $guest_pci_nr,virtio-net,$tap"
+  guest_pci_nr=$(expr $guest_pci_nr + 1)
+fi
+
+function launch_hard_rt_vm()
+{
+#for memsize setting
+mem_size=1024M
+
+# extract the guest kernel
+if [ ! -z $img ]; then
+  # mount virtual disk
+  losetup -f -P --show $img
+  mount /dev/loop0p3 /mnt
+
+  # extract the kernel
+  cp /mnt/boot/bzImage.preempt-rt /tmp/bzImage.preempt-rt
+
+  # umount disk
+  umount /mnt
+  losetup -D
+elif [ ! -z $disk ];then
+  # mount the physical disk
+  mkdir -p /mnt
+  mount $disk /mnt
+
+  # extract the kernel
+  cp /mnt/boot/bzImage.preempt-rt /tmp/bzImage.preempt-rt
```

Comment:
Would it be a viable alternative to mount that partition read-only and use it directly?

Not blocking this PR but I think it would be a better mechanism (assuming it works of course :-))